craft_application/grammar.py

```diff
@@ -55,14 +56,23 @@ def process_part(
         unprocessed_grammar = part_yaml_data[key]
         craft_cli.emit.debug(f"Processing grammar for {key}: {unprocessed_grammar}")
 
+        # ignore non-grammar keywords
+        if key not in get_grammar_aware_part_keywords():
+            continue
+
         # grammar aware models can be strings or list of dicts and strings
         if isinstance(unprocessed_grammar, list):
             # all items in the list must be a dict or a string
             if any(not isinstance(d, dict | str) for d in unprocessed_grammar):  # type: ignore[reportUnknownVariableType]
                 continue
+
             # all keys in the dictionary must be a string
-            if any(not isinstance(key, str) for d in unprocessed_grammar for key in d):  # type: ignore[reportUnknownVariableType]
```

Comment:
This was iterating through each character in a string, which was not causing any issues but was inefficient.